signingscript/sign.py

```diff
@@ -625,3 +637,67 @@ def _get_tarfile_compression(compression):
         return to
     except Exception as e:
         raise SigningScriptError(e)
+
+
+async def sign_file_with_autograph(context, from_, fmt, to=None):
+    """Signs a file with autograph and writes the result to arg `to` or `from_` if `to` is None.
+
+    Args:
+        context (Context): the signing context
+        from_ (str): the source file to sign
+        fmt (str): the format to sign with
+        to (str, optional): the target path to sign to. If None, overwrite
+            `from_`. Defaults to None.
+
+    Raises:
+        Requests.RequestException: on failure
+
+    Returns:
+        str: the path to the signed file
+
+    """
+    signer_type = task.task_signer_type(context)
+    scope = task.get_autograph_signer_scope(context, signer_type)
+
+    servers = get_suitable_signing_servers(context.signing_servers, scope, [fmt])
+    if not servers:
+        raise SigningScriptError(
+            "No signing servers found with scope {}".format(scope)
+        )
+    s = servers[0]
+    to = to or from_
+
+    with open(from_, 'rb') as fin:
+        input_bytes = fin.read()
+
+    # build and run the signature request
+    sign_req = [{"input": base64.b64encode(input_bytes)}]
```

Comment:
we may have some memory issues with this. ASAN updates are ~200MB, so this will be about 266MB encoded.

Reply:
Hm. Is there a way for us to use a generator or filehandle? Stack Overflow isn't cooperating for some reason :)
If not, our current instances can probably handle this; it looks like we sign a single file at a time.